01 OpenMW/Data Files/scripts/TamrielData/global.lua

```diff
@@ -0,0 +1,45 @@
+local types = require('openmw.types')
+local I = require('openmw.interfaces')
+
+-- Store the item record here
+local record
+
+-- Equip slots for the items we want to check
+local helmetSlot = types.NPC.EQUIPMENT_SLOT.Helmet
+local bootsSlot = types.NPC.EQUIPMENT_SLOT.Boots
+
+I.ItemUsage.addHandlerForType(types.Armor, function(armor, actor)
+    -- Sanity checks, then check if we're Imga
+    if not armor then return end
+    if types.NPC.record(actor).race ~= "T_Val_Imga" then return end
```

Comment:
`types.NPC.record(actor)` is going to fail for actors that aren't NPCs.

For NPCs, this check is always going to be true because `race` is a normalized/serialized ID. An Imga NPC would have a `race` of `t_val_imga`.

Reply:
Are you sure about the first point? It definitely works for player actors, since I tested the script by equipping the items on the player.

Reply:
Players are NPCs are Actors. It'd fail on creatures.

Reply:
This should be resolved by adding the `if not types.Player.objectIsInstance(actor) then return end` check.